frontend/src/main/components/Commons/Profits.js

```diff
@@ -2,23 +2,38 @@ import React from "react";
 import { Card } from "react-bootstrap";
 import ProfitsTable from "main/components/Commons/ProfitsTable"
 
-const dummyData = [
-    { id: 1, profit: 10, date: "2021-03-05" },
-    { id: 2, profit: 11, date: "2021-03-06" },
-    { id: 3, profit: 10, date: "2021-03-07" },
-    { id: 4, profit: 8, date: "2021-03-08" }
-];
+// const dummyData = [
+//     { id: 1, profit: 10, date: "2021-03-05" },
+//     { id: 2, profit: 11, date: "2021-03-06" },
+//     { id: 3, profit: 10, date: "2021-03-07" },
+//     { id: 4, profit: 8, date: "2021-03-08" }
+// ];
 
+// Helper function that accepts a timestamp and returns date as string
+function format_date(s) {
```

Comment:
Does it make more sense to use a date formatting library here instead of rolling our own? If we are rolling our own date format utilities, they should be extracted into a separate file and unit tested separately.

Reply:
I didn't want to import any libraries without express permission, and I figured I could just make a function since it's so simple. I'm just converting a timestamp into a date string using the built-in `Date` class, so there isn't very much code to be tested. If you'd like, I can call one of the built-in formatting functions which outputs an entire date string (I only did it this way to match the format of the provided "dummy" values). I could also keep what I have and factor it out with a few tests. Or, I could use some external library for this, but I generally try to avoid adding unnecessary dependencies. Please let me know how to proceed.

Reply:
Both options are good. I would personally lean toward using a date library like date-fns or moment for a few reasons:
- It allows the date format to be specified with a format string like `yyyy-MM-dd`. This is generally more readable and easier to change than composing the string manually.
- It allows us to write code closer to how we would write it with Java's robust built-in date support. JavaScript built-in dates are much more primitive to work with in comparsion.
- It can be particularly hard to handle edge cases when dealing with dates. Relying on a well-tested library means we can avoid a lot of these issues, plus we usually get better error handling from third-party libraries as well.
- Date manipulation is just a really common task on the frontend. We'll eventually need to parse dates and format dates based on whether they need to human-readable or machine-readable, and having a date library for that is super helpful.

Reply:
Refactoring the function and testing it separately is also a perfectly good option, it does save a dependency and it serves as an opportunity to do some pure unit testing. If you go this route, you should aim to be as thorough as possible when testing, both to catch edge cases and because testing serves as a form of documentation for your code.

Reply:
Per your recommendation, I have replaced my format function's implementation with a call to moment.js. Thanks for your insight!